app/src/main/java/com/mobileasone/dagger2workshop/di/AddNoteFragmentModule.kt

```diff
@@ -0,0 +1,27 @@
+package com.mobileasone.dagger2workshop.di
+
+import com.mobileasone.dagger2workshop.domain.repositories.NotesRepository
+import com.mobileasone.dagger2workshop.presentation.addnotes.AddNotePresenter
+import com.mobileasone.dagger2workshop.presentation.addnotes.AddNotePresenterImpl
+import com.mobileasone.dagger2workshop.util.ImageFile
+import com.mobileasone.dagger2workshop.util.ImageFileImpl
+import dagger.Module
+import dagger.Provides
+
+/**
+ * This module will provide dependencies for NoteListFragment
+ */
+@Module
+class AddNoteFragmentModule {
+
+    @Provides
+    fun imageFile(): ImageFile {
+        return ImageFileImpl()
+    }
+
+    @Provides
+    fun addNoteFragmentPresenter(repository: NotesRepository, imageFile: ImageFile): AddNotePresenter {
+        return AddNotePresenterImpl(repository, imageFile)
```

Comment:
One way to provide instances of AddNotePresenterImpl. 
We say how to create it so @Inject annotation is not needed with the constructor of [AddNotePresenterImpl.kt](https://github.com/Jamargle/Dagger2Workshop/pull/6/files#diff-87dd3aaebda8ea351ee26a554137f022).
Dagger will provide instances of the arguments looking for them in the Module class (ImageFile), the components or in the parent component if it exists (for getting the instance of 'NotesRepository' it will take it from SampleAppModule since AddNoteFragmentComponent is dependant on SampleAppModule).